capstone/src/test/java/com/google/sps/StudentServletTest.java

```diff
@@ -0,0 +1,177 @@
+package com.google.sps.servlets;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public final class StudentServletTest {
+  private static final String STUDENT = "student";
+  public static final String MEGHA_EMAIL = "kakm@google.com";
+  public static final String KEVIN_EMAIL = "kshao@google.com";
+  public static final String MEGAN_EMAIL = "meganshi@google.com";
+  public static final String MEGAN_NAME = "Megan Shi";
+  public static final String MEGHA_NAME = "Megha Kak";
+  public static final int YEAR_2022 = 2022;
+  public static final String MAJOR = "Computer Science";
+  public static final String CLUB_1 = "Club 1";
```

Comment:
Quick question: are these constants specific to the individual test use cases? If so, it's actually better to define them or even hard-code them in the specific test. Test constants are generally not used except for commonly shared cases across multiple tests (ex: the email of the user is always going to return the same thing if we're using the same user in all the tests). 

If it's for set-up though, it's ok to leave it as-is.

Reply:
These are for set up so we can initially have entities in Datastore to test. The only one that isn't for set up is STUDENT - however, this is used in all of the tests to access the JSON content. I don't think any of these are used for specific test cases, but I will keep this in mind for the future :)